lib/pyfrc/test_support/pytest_plugin.py

```diff
@@ -59,6 +59,9 @@ def pytest_runtest_setup(self):
                        "If your robot class has an __init__ function, it must call super().__init__()!"
     
     def pytest_runtest_teardown(self, nextitem):
+        # Let any child threads run in realtime to allow cancelling if it
+        # has been implemented.
+        self._fake_time.freerun_children()
```

Comment:
Yes, that seems fine to me, provided that there's a way that a thread can still successfully shutdown.

Now the question becomes -- what is the best way for someone to detect that the test has ended and tear themselves down? I suppose a user could do it with a try... finally block in their test. For things that are inside WPILib, we use the `Resource` class. It's not public API, but we could make it so.